frontend/src/components/CheckboxGroup/CheckboxGroupValidator.tsx

```diff
@@ -0,0 +1,221 @@
+import { List } from 'immutable';
+import React, { FunctionComponent, useState } from 'react';
+import { Filters, OperationStepMap } from '../../types/operations';
+import { BasicModal } from '../BasicModal';
+
+interface ComponentProps {
+  steps?: List<OperationStepMap>;
+}
+
+const CheckboxGroupValidator: FunctionComponent<ComponentProps> = (props) => {
+  const [showModal, setShowModal] = useState<boolean>(false);
+  const [modalMessage, setModalMessage] = useState<string>('');
+
+  const validateCheckboxOptions = (
+    checkboxValue: string,
+    checkboxState: boolean | undefined,
+  ): void => {
+    const { steps } = props;
+    steps &&
+      steps.map((step) => {
+        const options = step.get('query_kwargs') as string;
+        toggleModalState(options, checkboxState, checkboxValue);
+      });
```

Comment:
Use an `if` statement.

Reply:
This seems like the place to use the active step prop to filter these steps so that you only check those that come AFTER our active step.

Reply:
Updated and added filter to exclude active step